src/pkg/services/m365/api/item_pager.go

```diff
@@ -230,3 +240,33 @@ func addedAndRemovedByAddtlData[T any](items []T) ([]string, []string, error) {
 
 	return added, removed, nil
 }
+
+// for added and removed by GetDeletedDateTime()
+
+type getIDAndDeletedDateTimer interface {
+	getIDer
+	GetDeletedDateTime() *time.Time
+}
+
+func addedAndRemovedByDeletedDateTime[T any](items []T) ([]string, []string, error) {
+	added, removed := []string{}, []string{}
+
+	for _, item := range items {
+		giaddt, ok := any(item).(getIDAndDeletedDateTimer)
+		if !ok {
+			return nil, nil, clues.New("item does not provide id and deleted date time getters").
+				With("item_type", fmt.Sprintf("%T", item))
+		}
+
+		// if the additional data contains a `@removed` key, the value will either
+		// be 'changed' or 'deleted'.  We don't really care about the cause: both
+		// cases are handled the same way in storage.
```

Comment:
nit: could you please update this comment block to match the code below it?

Reply:
whoops, can do.